overlord/snapstate/aliasesv2_test.go

```diff
@@ -0,0 +1,402 @@
+		"alias4": {Status: "auto", Target: "cmd4"},
+	})
+
+	new, err = snapstate.RefreshAliasStates(s.state, info, map[string]*snapstate.AliasState{
+		"alias1":  {Status: "disabled", Target: "cmd1old"},
+		"alias5":  {Status: "disabled", Target: "cmd5"},
+		"alias6":  {Status: "disabled", Target: "cmd6"},
+		"alias4":  {Status: "overridden", Target: "cmd3", AutoTargetBak: "cmd4"},
+		"manual3": {Status: "manual", Target: "cmd3"},
+		"manual7": {Status: "manual", Target: "cmd7"},
+	})
+	c.Assert(err, IsNil)
+	c.Check(new, DeepEquals, map[string]*snapstate.AliasState{
+		"alias1":  {Status: "disabled", Target: "cmd1"},
+		"alias2":  {Status: "disabled", Target: "cmd2"},
+		"alias4":  {Status: "overridden", Target: "cmd3", AutoTargetBak: "cmd4"},
+		"manual3": {Status: "manual", Target: "cmd3"},
+	})
+}
+
+func (s *snapmgrTestSuite) TestCheckAliasStatesConflictsAgainstAliases(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	s.state.Set("aliases-v2", map[string]map[string]*snapstate.AliasState{
+		"other-snap1": {
+			"alias1": {Status: "disabled", Target: "cmd1"},
+			"alias2": {Status: "disabled", Target: "cmd2"},
+		},
+		"other-snap2": {
+			"alias2": {Status: "auto", Target: "cmd2"},
+			"alias3": {Status: "auto", Target: "cmd3"},
+		},
+		"other-snap3": {
+			"alias4": {Status: "manual", Target: "cmd8"},
+			"alias5": {Status: "disabled", Target: "cmd5"},
+		},
+	})
+
+	confl, err := snapstate.CheckAliasStatesConflicts(s.state, "alias-snap", map[string]*snapstate.AliasState{
+		"alias1": {Status: "auto", Target: "cmd1"},
+		"alias5": {Status: "auto", Target: "cmd5"},
+	})
+	c.Check(err, IsNil)
+	c.Check(confl, IsNil)
+
+	confl, err = snapstate.CheckAliasStatesConflicts(s.state, "alias-snap", map[string]*snapstate.AliasState{
+		"alias1": {Status: "auto", Target: "cmd1"},
+		"alias2": {Status: "auto", Target: "cmd2"},
+		"alias3": {Status: "auto", Target: "cmd3"},
+		"alias4": {Status: "auto", Target: "cmd4"},
+	})
+	c.Check(err, FitsTypeOf, &snapstate.AliasConflictError{})
+	c.Check(confl, HasLen, 2)
+	which := confl["other-snap2"]
+	sort.Strings(which)
+	c.Check(which, DeepEquals, []string{"alias2", "alias3"})
+	c.Check(confl["other-snap3"], DeepEquals, []string{"alias4"})
+}
+
+func (s *snapmgrTestSuite) TestAliasConflictError(c *C) {
+	e := &snapstate.AliasConflictError{Snap: "foo", Conflicts: map[string][]string{
+		"bar": {"baz"},
+	}}
+	c.Check(e, ErrorMatches, `cannot enable alias "baz" for "foo", already enabled for "bar"`)
+
+	e = &snapstate.AliasConflictError{Snap: "foo", Conflicts: map[string][]string{
+		"bar": {"baz1", "baz2"},
+	}}
+	c.Check(e, ErrorMatches, `cannot enable aliases "baz1", "baz2" for "foo", already enabled for "bar"`)
+
+	e = &snapstate.AliasConflictError{Snap: "foo", Conflicts: map[string][]string{
+		"bar1": {"baz1"},
+		"bar2": {"baz2"},
+	}}
+	c.Check(e, ErrorMatches, `cannot enable alias "baz." for "foo", already enabled for "bar." nor alias "baz." already enabled for "bar."`)
```

Comment:
these all feel like the sort of thing where explaining them to the user is going to be hard, and giving the user a next step to follow to resolve it even harder, without structured information.